pom.xml

```diff
@@ -280,8 +280,8 @@
             <groupId>org.jenkins-ci.tools</groupId>
             <artifactId>maven-hpi-plugin</artifactId>
             <configuration>
-              <!-- JENKINS-50520: since we need a custom version of Guava -->
-              <maskClasses>com.google.common.</maskClasses>
+              <!-- JENKINS-50520: since we need a custom version of Guava and Guice -->
```

Comment:
you really don't want to go here - it does not work as demonstrated with jacoco. (it works in the plugin but depending on the Guice initialisation and what other plugins you have it will randomly blow up during initialisation for some users...)

Reply:
Why do you think that? Whatever jclouds is using Guice for, it should be isolated from the stuff that Jenkins core and other plugins are using it for (extension loading).

What I would really like to do here is remove the jclouds dependency altogether, and rewrite the implementation to use the AWS SDK exclusively (it already has to use it to get presigned URL support for blobs, which jclouds does not support). The original idea had been to use jclouds as the multiplexing layer so that we could rapidly add support for lots of blob stores, but things never worked out that way (the Azure folks went with their own implementation), and it has been a source of frustration and opacity.